### PR TITLE
Add .gitignore

### PR DESCRIPTION
**Pull request by joanhey**
Add  all `vendor/` dirs to `.gitignore`.
Without when we setup the frameworks, exist +4k files to track in Git and it's unmanageable.

Still so, exist ~1k files to track, that complicate to manage the changes.

I think that it will be good to change the dirs structure, like  `/framework-xx/code/`  (code, bench, ... you can choose).
So we can add to `.gitignore` too `/*/code/` and we will manage the changes only in the important files. 
At the same time is easier to delete (clean.sh) and we don't need the temporal dir `temp` in _benchmark to later copy the files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/output/
```
Comment (myaaghubi) [inline .gitignore:1]: Already have a `.gitignore` inside of this dir

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+vendor/
```
Comment (myaaghubi) [inline .gitignore:2]: I didn't notice the `vendor` dir in the root, maybe because of a different configuration of the composer! so it is accepted

**Comment by myaaghubi** [comment]
Check out the notes on you commit.
Another idea to change the structure is to have a dir for pre-configured files (for frameworks) to completely separate them from what the `composer` gonna clone then we not gonna have the `temp` dir and stuff gonna goes easier for `clean.sh` as you mentioned and even may not mention the version number on related folders too! 
However at the moment I prefer the old structure :sleepy:

File: .gitignore (hunk 2 above)
Comment (joanhey) [reply .gitignore:2]: `/dir/` is any dir that start at the root of git
`vendor/` is any dir with name vendor in all the repo, so any `framework-xx/vendor/` and /**

**Comment by joanhey** [comment]
image

**Comment by joanhey** [comment]
Also like anybody can update the frameworks via composer, it's unnecessary to have the minor versions in the framework dir.

It's enough with Codeigniter4 or Codeigniter3, ....

**Pull request merged**